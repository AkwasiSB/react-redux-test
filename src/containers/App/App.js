import React, { Fragment, Suspense } from 'react';
import ErrorBoundary from '../../components/ErrorBoundary/ErrorBoundary';
import Header from '../../components/Header/Header';
import Footer from '../../components/Footer/Footer';
import PostForm from '../PostForm/PostForm';

const Posts = React.lazy(() => import('../Posts/Posts'));

const App = () => (
	<div>
		<ErrorBoundary>
			<Fragment>
				<Header title='Posts' />

				<main>
					<PostForm />

					<Suspense fallback={<div>Loading...</div>}>
						<Posts />
					</Suspense>
				</main>

				<Footer />
			</Fragment>
		</ErrorBoundary>
	</div>
);

export default App;
