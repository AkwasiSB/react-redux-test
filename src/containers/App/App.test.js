import App from './App';

test('if App renders correctly', () => {
    const appWrapper = shallow(<App />);

    expect(appWrapper).toMatchSnapshot();
})
