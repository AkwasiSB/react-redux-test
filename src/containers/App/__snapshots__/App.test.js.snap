// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if App renders correctly 1`] = `
<div>
  <ErrorBoundary>
    <Header
      title="Posts"
    />
    <main>
      <Connect(PostForm) />
      <React.Suspense
        fallback={
          <div>
            Loading...
          </div>
        }
      >
        <React.Lazy />
      </React.Suspense>
    </main>
    <Footer />
  </ErrorBoundary>
</div>
`;
